src/main/kotlin/com/autonomousapps/tasks/AdviceTask.kt

```diff
@@ -132,58 +137,85 @@ abstract class AdviceTask : DefaultTask() {
+    chatter.chat("See pretty report at ${advicePrettyFile.path}")
+    chatter.chat("See machine-readable console report at ${adviceConsoleFile.path}")
+    chatter.chat("See pretty console report at ${adviceConsolePrettyFile.path}")
 
-    if (!computedAdvice.filterChange) {
-      advicePrinter.getChangeAdvice()?.let {
-        consoleReport.add("Existing dependencies which should be modified " +
-          "to be as indicated:\n$it\n")
-        didGiveAdvice = true
-      }
+    if (consoleReport.isEmpty()) {
+      chatter.chat("Looking good! No changes needed")
+    } else {
+      chatter.chat("See console report at ${adviceConsoleFile.path}")
     }
+  }
 
-    if (!computedAdvice.filterCompileOnly) {
-      advicePrinter.getCompileOnlyAdvice()?.let {
-        consoleReport.add("Dependencies which could be compile-only:\n$it\n")
-        didGiveAdvice = true
+  internal data class ConsoleReport(
+    val addToApiAdvice: Set<Advice> = emptySet(),
+    val addToImplAdvice: Set<Advice> = emptySet(),
+    val removeAdvice: Set<Advice> = emptySet(),
+    val changeToApiAdvice: Set<Advice> = emptySet(),
+    val changeToImplAdvice: Set<Advice> = emptySet(),
+    val compileOnlyDependencies: Set<Dependency> = emptySet(),
+    val unusedProcsAdvice: Set<Advice> = emptySet()) {
+
+    fun isEmpty() = addToApiAdvice.isEmpty() &&
+      removeAdvice.isEmpty() &&
+      changeToApiAdvice.isEmpty() &&
+      changeToImplAdvice.isEmpty() &&
+      compileOnlyDependencies.isEmpty() &&
+      unusedProcsAdvice.isEmpty()
+
+    companion object {
+      fun build(computedAdvice: ComputedAdvice): ConsoleReport {
+        lateinit var addToApiAdvice: Set<Advice>
+        lateinit var addToImplAdvice: Set<Advice>
+        lateinit var removeAdvice: Set<Advice>
+        lateinit var changeToApiAdvice: Set<Advice>
+        lateinit var changeToImplAdvice: Set<Advice>
+        lateinit var compileOnlyDependencies: Set<Dependency>
+        lateinit var unusedProcsAdvice: Set<Advice>
+
+        if (!computedAdvice.filterRemove && computedAdvice.removeAdvice.isNotEmpty()) {
+          removeAdvice = computedAdvice.removeAdvice
+        }
+
+        val addAdvices = computedAdvice.addToApiAdvice + computedAdvice.addToImplAdvice
+        if (!computedAdvice.filterAdd && addAdvices.isNotEmpty()) {
+          addToApiAdvice = computedAdvice.addToApiAdvice
+          addToImplAdvice = computedAdvice.addToImplAdvice
+        }
+
+        val changeAdvices = computedAdvice.changeToApiAdvice + computedAdvice.changeToImplAdvice
+        if (!computedAdvice.filterChange && changeAdvices.isNotEmpty()) {
+          changeToApiAdvice = computedAdvice.changeToApiAdvice
+          changeToImplAdvice = computedAdvice.changeToImplAdvice
+        }
+
+        if (!computedAdvice.filterCompileOnly && computedAdvice.compileOnlyDependencies.isNotEmpty()) {
+          compileOnlyDependencies = computedAdvice.compileOnlyDependencies
+        }
+
+        // TODO add filter
```

Comment:
If the TODO is just about making a filter in the same way as others, without anything specific, I can do that.

Reply:
Yes, it's about making a filter in the same way as the other filters (trickling down from `DependencyAnalysisExtension`, but I wouldn't bother with that in this PR. It seems increasingly likely I'm going to rewrite how filtering / customization works.

Reply:
Let see how it goes to get the renamer feature done. I create a ticket for it.